.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,2093 @@
+      shell: bash
+      run: |
+        cd OpenDDS
+        . setenv.sh
+        mwc.pl -type gnuace $TAO_ROOT/tests/Hello
+    - name: build TAO Hello test
+      shell: bash
+      run: |
+        cd OpenDDS
+        . setenv.sh
+        make -j2 dir=$TAO_ROOT/tests/Hello
+    - name: set up Modeling tests
+      shell: bash
+      run: |
+        cd OpenDDS
+        . setenv.sh
+        cd tools/modeling/tests
+        ./setup.pl
+        cd -
+        mwc.pl -type gnuace -features built_in_topics=0 tools/modeling/tests/modeling_tests.mwc
+    - name: build Modeling tests
+      shell: bash
+      run: |
+        cd OpenDDS
+        . setenv.sh
+        make -j2 -C tools/modeling/tests
+    - name: run OpenDDS tests
+      shell: bash
+      run: |
+        cd OpenDDS
+        . setenv.sh
+        mkdir workspace
+        cd workspace
+        cat > config.xml <<EOF
+        <autobuild>
+        <configuration>
+        <variable name="log_file" value="ubuntu-16_04_java_no-bit_autobuild_output.log"/>
+        <variable name="log_root" value="$DDS_ROOT/workspace"/>
+        <variable name="project_root" value="$DDS_ROOT"/>
+        <variable name="root" value="$DDS_ROOT/workspace"/>
+        <variable name="junit_xml_output" value="Tests"/>
+        </configuration>
+        <command name="log" options="on"/>
+        <command name="git" options="dir=$GITHUB_WORKSPACE/OpenDDS log -1"/>
+        <command name="print_os_version"/>
+        <command name="print_env_vars"/>
+        <command name="print_ace_config"/>
+        <command name="print_make_version"/>
+        <command name="check_compiler" options="gcc"/>
+        <command name="print_autobuild_config"/>
+        <command name="auto_run_tests" options="dir=$GITHUB_WORKSPACE/OpenDDS -Config RTPS -Config RAPIDJSON -Config NO_BUILT_IN_TOPICS -a -l java/tests/dcps_java_tests.lst -l tools/modeling/tests/modeling_tests.lst"/>
+        <command name="log" options="off"/>
+        <command name="process_logs" options="prettify"/>
+        </autobuild>
+        EOF
+        $GITHUB_WORKSPACE/autobuild/autobuild.pl "$GITHUB_WORKSPACE/OpenDDS/workspace/config.xml"
+    - name: upload Full log output
+      uses: actions/upload-artifact@v2
+      with:
+        name: ubuntu-16_04_java_no-bit_autobuild_output.log_Full.html
+        path: OpenDDS/workspace/ubuntu-16_04_java_no-bit_autobuild_output.log_Full.html
+    - name: check results
+      shell: bash
+      run: |
+        cat "$GITHUB_WORKSPACE/OpenDDS/workspace/latest.txt"
+        grep -q 'Failures: 0' "$GITHUB_WORKSPACE/OpenDDS/workspace/latest.txt"
+    - name: Publish Unit Test Results
+      uses: simpsont-oci/action-junit-report@v1.1.2
+      if: always()
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        check_name: "Test Results: ubuntu-16_04_java_no-bit"
+        report_paths: OpenDDS/workspace/Tests_JUnit.xml
+
+  # Windows 2019
+
+  windows-2019_defaults_artifact:
+
+    runs-on: windows-2019
```

Comment:
why duplicate everything for windows-2016 or windows-2019, using a matrix there is less duplication, see for example https://github.com/RemedyIT/taox11/blob/454f313787885b345a26a3960bcb49abb76ec432/.github/workflows/windows.yml#L27